util/test/timers_tests.cpp

```diff
@@ -0,0 +1,98 @@
+// Concord
+//
+// Copyright (c) 2019 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+//
+
+#include <cstdlib>
+#include "gtest/gtest.h"
+#include "Timers.hpp"
+
+using namespace std;
+using namespace std::chrono;
+
+namespace concordUtil {
+
+TEST(TimersTest, Basic) {
+  milliseconds duration(100);
+
+  int oneshot_counter = 0;
+  int recurring_counter = 0;
+  auto heap = TimerHeap();
+
+  // Create a oneshot and a recurring timer and add them to the heap. Each one
+  // will fire a callback when it expires that increments a counter.
+  heap.emplace(Timer(duration, Timer::ONESHOT, [&oneshot_counter]() { ++oneshot_counter; }));
+  auto recurring_handle = heap.emplace(
+      Timer(duration, Timer::RECURRING, [&recurring_counter]() { ++recurring_counter; }));
+
+  steady_clock::time_point now = steady_clock::now();
```

Comment:
If it takes 100ms to get from line 33 to here, things are going very poorly, but would it be better to move this line above the two `Timer` instantiations anyway? They each call `steady_clock::now` as well, and reordering would make sure that the `now` passed to `evaluate` on line 40 would not have advanced.

Reply:
Nice catch! I changed this in a different manner. There is now a Timer constructor that takes the now as a parameter, like the other functions. This allows tests to be deterministic.